sys-devel/gcc/gcc-5.4.0_2016_07_08.recipe

```diff
@@ -0,0 +1,347 @@
+SUMMARY="C/C++ compiler for target ${effectiveTargetMachineTriple}"
+DESCRIPTION="The standard compiler for non-legacy Haiku (i.e. for all \
+architectures other than x86_gcc2)."
+HOMEPAGE="http://gcc.gnu.org"
+
+SOURCE_URI="https://ftp.gnu.org/gnu/gcc/gcc-5.4.0/gcc-5.4.0.tar.bz2"
```

Comment:
There is the "safe source" problem, but HTTP is considered "safe" if there is a checksum (we can check we got the right file). What's "unsafe" is downloading from eg. a git repo, where you can't checksum anything (there isn't a single file, but several ones).

I'm not sure https on mirrors is very useful, because:
1) It would redirect to somewhere else anyway
2) You are redirected to a random website, and get a matching certificate. That does not mean you trust whatever you have been redirected to?

Using the official mirror from GNU for the initial writing of the recipe is a good idea, because it makes sure we get the official binary and the correct SHA256. But once the recipe is written, switching to the mirrors is ok, we just need to be careful when updating the recipe, that the SHA256 always matches what's on GNU main website (so we don't get a corrupted/tainted file from a mirror and think it is the "official" source).